lib/AST/ASTPrinter.cpp

```diff
@@ -6387,10 +6387,27 @@ class TypePrinter : public TypeVisitor<TypePrinter> {
 
       // Print based on the type.
       Printer << "some ";
-      if (auto inheritedType = decl->getInherited().front().getType())
-        inheritedType->print(Printer, Options);
-      else
-        Printer << "Any";
+      if (!decl->getConformingProtocols().empty()) {
+        ArrayRef<ProtocolDecl *> protocols = decl->getConformingProtocols();
+        bool printed = false;
+        for (unsigned i = 0, e = protocols.size(); i < e; i++) {
```

Comment:
You can use LLVM's `interleave` to do this well, and it'll handle when the print the separator and such.

Reply:
Good call! I've pushed up a commit that switches to this.